thrust/thrust/system/cuda/detail/sort.h

```diff
@@ -299,7 +299,14 @@ namespace __smart_sort
 template <class Key, class CompareOp>
 using can_use_primitive_sort = ::cuda::std::integral_constant<
   bool,
-  ::cuda::std::is_arithmetic<Key>::value
+  (::cuda::std::is_arithmetic<Key>::value
+#  ifdef _CCCL_HAS_NVFP16
```

Comment:
Critical: This assumes that `__half` and `__nvbfloat16` behave as arithmetic types, but a user might have the macros like `__CUDA_NO_HALF_OPERATORS__`

I believe we need to guard against any of those macros being defined

Reply:
Added those guards. Maybe radix sort would work without some operators or conversions, but let's be conservative.